CRM/EventsExtras/Hook/BuildForm/EventFee.php

```diff
@@ -34,15 +33,55 @@ private function buildForm($formName, &$form) {
 
   private function setDefaults(&$form) {
     $defaults = [];
-    $paymentProcessor = SettingsManager::SETTING_FIELDS['PAYMENT_PROCESSOR_SELECTION'];
+    $fieldIdsToHide = [];
+
+    $showPaymentProcessor = SettingsManager::SETTING_FIELDS['PAYMENT_PROCESSOR_SELECTION'];
     $paymentProcessorDefault = SettingsManager::SETTING_FIELDS['PAYMENT_PROCESSOR_SELECTION_DEFAULT'];
-    $settings = [$paymentProcessor, $paymentProcessorDefault];
+    $settings = [$showPaymentProcessor, $paymentProcessorDefault];
     $settingValues = SettingsManager::getSettingsValue($settings);
-    if ($settingValues[$paymentProcessor] == 0) {
+    if ($settingValues[$showPaymentProcessor] == 0) {
       $defaultSettingString = implode(CRM_Core_DAO::VALUE_SEPARATOR, $settingValues[$paymentProcessorDefault]);
       $paymentProcessorDefaultValue = (array_fill_keys(explode(CRM_Core_DAO::VALUE_SEPARATOR, $defaultSettingString), '1'));
       $defaults['payment_processor'] = $paymentProcessorDefaultValue;
+      $fieldIdsToHide[] = 'payment_processor';
+    }
+
+    $showCurrency = SettingsManager::SETTING_FIELDS['CURRENCY'];
+    $currencyDefault = SettingsManager::SETTING_FIELDS['CURRENCY_DEFAULT'];
+    $settings = [$showCurrency, $currencyDefault];
+    $settingValues = SettingsManager::getSettingsValue($settings);
+    if ($settingValues[$showCurrency] == 0) {
+      $defaults['currency'] = $settingValues[$currencyDefault];
+      $fieldIdsToHide[] = 'currency';
+    }
+
+    $showPayLater = SettingsManager::SETTING_FIELDS['PAY_LATER_OPTION'];
+    $payLaterDefault = SettingsManager::SETTING_FIELDS['PAY_LATER_OPTION_DEFAULT'];
+    $payLaterLabel = SettingsManager::SETTING_FIELDS['PAY_LATER_OPTION_DEFAULT_LABEL'];
+    $payLaterInstruction = SettingsManager::SETTING_FIELDS['PAY_LATER_OPTION_DEFAULT_LABEL_INSTRUCTION'];
+    $payLaterBillingAddress = SettingsManager::SETTING_FIELDS['PAY_LATER_OPTION_DEFAULT_BILLING_ADDRESS'];
+    $settings = [$showPayLater, $payLaterDefault, $payLaterLabel, $payLaterInstruction, $payLaterBillingAddress];
+    $settingValues = SettingsManager::getSettingsValue($settings);
+    if ($settingValues[$showPayLater] == 0) {
+      $defaults['is_pay_later'] = $settingValues[$payLaterDefault];
+      $defaults['pay_later_text'] = $settingValues[$payLaterLabel];
+      $defaults['pay_later_receipt'] = $settingValues[$payLaterInstruction];
+      $defaults['is_billing_required'] = $settingValues[$payLaterBillingAddress];
+      $fieldIdsToHide[] = 'is_pay_later';
+      $fieldIdsToHide[] = 'pay_later_text';
+      $fieldIdsToHide[] = 'pay_later_receipt';
+
+      // @note is_billing_required's parent tr in civicrm/templates/CRM/Event/Form/ManageEvent/Registration.tpl
+      // doesn't have any CSS class (CiviCRM bug) as a workaround we could use another selector
+      // $fieldIdsToHide[] = 'is_billing_required';
+      $this->hideElements('#payLaterOptions tr:nth-child(5)');
```

Comment:
as we discussed in the call, this won't be needed since you are hiding the whole parent below, also it seems you don't even need these lines above : 

```
$fieldIdsToHide[] = 'is_pay_later';
$fieldIdsToHide[] = 'pay_later_text';
$fieldIdsToHide[] = 'pay_later_receipt';
```

since again, you are hiding the whole parent

Reply:
I only needed the first one because it is not a child of the element `tr#payLaterOptions`

    $fieldIdsToHide[] = 'is_pay_later';